src/lib/context.ts

```diff
@@ -1,45 +1,67 @@
 import { Config } from "@oclif/core";
-import * as fs from "fs/promises";
-import * as path from "path";
+import { TerraformContextProvider } from "./context_terraform.js";
+import { UserContextProvider } from "./context_user.js";
+
+export type ContextNames = "project" | "server" | "org" | "installation";
+export type ContextKey<N extends ContextNames = ContextNames> = `${N}-id`;
```

Comment:
Falls du die Names nicht nur als Type sondern auch als String-Array brauchst, kannst du das so machen. Weiß nicht ob es dir hier wegen dem Template Literal Type was nutzt, aber vielleicht für später mal.

```suggestion
export const contextNames = ["project", "server", "org", "installation"] as const;
export type ContextNames = keyof typeof contextNames;
export type ContextKey<N extends ContextNames = ContextNames> = `${N}-id`;
```

Reply:
Tatsächlich sorgt das bei mir für einen ganzen Haufen Folgefehler, unter anderem bei der Definition von `ContextKey`:

```
src/lib/context.ts:12:68 - error TS2322: Type 'N' is not assignable to type 'string | number | bigint | boolean | null | undefined'.
  Type 'keyof readonly ["project", "server", "org", "installation"]' is not assignable to type 'string | number | bigint | boolean | null | undefined'.
    Type 'unique symbol' is not assignable to type 'string | number | bigint | boolean | null | undefined'.

12 export type ContextKey<N extends ContextNames = ContextNames> = `${N}-id`;
                                                                      ~
```

Brauchen würde ich die `contextNames`-Konstante hier nicht.